### PR TITLE
Add support of `$RANDOM` token in smoke test config.

### PR DESCRIPTION
**Pull request by shinfan**
- Fixed an issue where the smoke test data was not init correctly.

### CODE REVIEW

File: src/main/java/com/google/api/codegen/metacode/InitFieldConfig.java [hunk 1]
```diff
@@ -23,6 +23,8 @@
  */
 @AutoValue
 public abstract class InitFieldConfig {
+  private static String randomValueToken = "$RANDOM";
```
Comment (michaelbausor) [inline src/main/java/com/google/api/codegen/metacode/InitFieldConfig.java]: Should this be final?

File: src/main/java/com/google/api/codegen/metacode/InitFieldConfig.java [hunk 2]
```diff
@@ -67,4 +69,9 @@ public boolean isFormattedConfig() {
   public boolean hasFormattedInitValue() {
     return entityName() != null && value() != null;
   }
+
+  private static String parseValueString(String valueString) {
+    String randomValue = Integer.toString(Math.abs(valueString.hashCode()));
```
Comment (michaelbausor) [inline src/main/java/com/google/api/codegen/metacode/InitFieldConfig.java]: This approach will produce the same "random" value for the same string. Is that what we want? The most common case will be just the string "$RANDOM", which is likely to occur often and will always produce the same value.

Is this a problem? Or is it more important for the value to be reproducible?

File: src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java [hunk 3]
```diff
@@ -329,15 +317,27 @@ private InitValueView getInitValue(MethodTransformerContext context, InitCodeNod
...
-      List<String> varList, InitValueConfig initValueConfig) {
+      MethodTransformerContext context, Field field, InitValueConfig initValueConfig) {
     List<String> formatFunctionArgs = new ArrayList<>();
-    for (String entityName : varList) {
+
+    Set<String> entityNames;
+    boolean shouldUseCollectionConfigEntityNames =
+        !context.getFeatureConfig().useResourceNameFormatOption(field)
+            || initValueConfig.hasFormattingConfigInitialValues();
+    if (shouldUseCollectionConfigEntityNames) {
```
Comment (michaelbausor) [inline src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java]: I found where the "_id" suffix is coming from:
https://github.com/googleapis/toolkit/blob/master/src/test/java/com/google/api/codegen/testdata/library.proto#L30

This is a problem that will be fixed once the resource name configuration moves out of the proto and into the gapic config, although it was a mistake to allow this difference in the first place - my original PR should have checked for this inconsistency.

In this PR, which do you think is preferable - the changes you have here, or switching shelf_id->shelf in the proto, or both?

Comment (shinfan) [inline src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java]: As discussed offline, I will change the proto and revert the fix in the java code.

File: src/main/java/com/google/api/codegen/metacode/InitFieldConfig.java (hunk 2 above)
Comment (shinfan) [inline src/main/java/com/google/api/codegen/metacode/InitFieldConfig.java]: Good point. I changed the hash string so the $RANDOM value would be more unique. It is not a strict requirements that each $RANDOM value needs to be unique. But it should be unique enough so the data can be validated by tests.

**Comment by shinfan** [comment]
@michaelbausor  PTAL

File: src/test/java/com/google/api/codegen/testdata/nodejs_doc_main_library.baseline [hunk 4]
```diff
@@ -180,78 +180,78 @@ function LibraryServiceApi(gaxGrpc, grpcClients, opts) {
...
  * Returns a fully-qualified shelf resource name string.
- * @param {String} shelf
+ * @param {String} shelf_id
  * @returns {String}
  */
-LibraryServiceApi.prototype.shelfPath = function(shelf) {
+LibraryServiceApi.prototype.shelfPath = function(shelfId) {
   return SHELF_PATH_TEMPLATE.render({
-    shelf: shelf
+    shelf_id: shelfId
```
Comment (garrettjonesgoogle) [inline src/test/java/com/google/api/codegen/testdata/nodejs_doc_main_library.baseline]: It looks like there is a casing bug in node js... the doc doesn't match the real parameter.

File: src/main/java/com/google/api/codegen/metacode/InitFieldConfig.java [hunk 5]
```diff
@@ -67,4 +69,9 @@ public boolean isFormattedConfig() {
   public boolean hasFormattedInitValue() {
     return entityName() != null && value() != null;
   }
+
+  private static String parseValueString(String valueString, String stringToHash) {
+    String randomValue = Integer.toString(Math.abs(stringToHash.hashCode()));
+    return valueString.replace(randomValueToken, randomValue);
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/metacode/InitFieldConfig.java]: We are generating a random number every time, even if we hardly ever do the replacement. Maybe we should do a find first?

File: src/test/java/com/google/api/codegen/testdata/csharp_gapic_client_library.baseline [hunk 6]
```diff
@@ -704,57 +704,57 @@ namespace Google.Example.Library.V1
...
         /// </list>
         /// </summary>
-        public static PathTemplate ShelfTemplate { get; } = new PathTemplate("shelves/{shelf}");
+        public static PathTemplate ShelfTemplate { get; } = new PathTemplate("shelves/{shelf_id}");
 
         /// <summary>
         /// Creates a shelf resource name from its component IDs.
         /// </summary>
-        /// <param name="shelfId">The shelf ID.</param>
+        /// <param name="shelfIdId">The shelfId ID.</param>
```
Comment (garrettjonesgoogle) [inline src/test/java/com/google/api/codegen/testdata/csharp_gapic_client_library.baseline]: CSharp is acting weird here.

Comment (shinfan) [inline src/test/java/com/google/api/codegen/testdata/csharp_gapic_client_library.baseline]: Issue filed: https://github.com/googleapis/toolkit/issues/644

File: src/main/java/com/google/api/codegen/metacode/InitFieldConfig.java (hunk 5 above)
Comment (shinfan) [inline src/main/java/com/google/api/codegen/metacode/InitFieldConfig.java]: Good point. Done.

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**